src/ray/raylet/worker.cc

```diff
@@ -43,6 +43,19 @@ const std::shared_ptr<LocalClientConnection> Worker::Connection() const {
   return connection_;
 }
 
+const ResourceIdSet &Worker::GetResourceIds() const {
+  return resource_ids_;
+}
+
+void Worker::ResetResourceIds() {
+  resource_ids_.clear();
+}
+
+void Worker::SetResourceIds(ResourceIdSet &resource_ids) {
+  RAY_CHECK(resource_ids_.size() == 0);
+  resource_ids_ = resource_ids;
```

Comment:
resource_ids_.swap(resource_ids); 
swap will swap the pointer of two containers, it is better than copy in the case.